### PR TITLE
Implemented the option to choose between ascending and descending on list calls

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Describe changes
You can now specify sort_by=asc:created or sort_by=desc:created on the list endpoints.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/models/filter_models.py [hunk 1]
```diff
@@ -278,12 +278,26 @@ class BaseFilterModel(BaseModel):
...
+        split_value = v.split(":", 1)
+        if (
+                len(split_value) == 2
+        ):
+            column = split_value[1]
+
+            if split_value[0] not in SorterOps.values():
+                logger.warning("Invalid operand used for column sorting."
+                               "Only 'asc' and 'desc' are supported."
+                               "Defaulting to 'asc' on column `%s`.",
```
Comment (schustmi) [inline src/zenml/models/filter_models.py:292]: Maybe use `SorterOps.values()` in the warning message as well to cover future extensions?

File: src/zenml/models/filter_models.py [hunk 2]
```diff
@@ -303,6 +317,26 @@ def list_of_filters(self) -> List[Filter]:
...
+        operator = 'asc'
+
+        # Check if user explicitly set an operand
+        split_value = self.sort_by.split(":", 1)
+        if (
+                len(split_value) == 2
+        ):
+            column = split_value[1]
+
+            if split_value[0] in SorterOps.values():
```
Comment (schustmi) [inline src/zenml/models/filter_models.py:334]: Isn't this already covered in the validator?

File: src/zenml/models/filter_models.py [hunk 3]
```diff
@@ -303,6 +317,26 @@ def list_of_filters(self) -> List[Filter]:
             {key: getattr(self, key) for key in self.__fields__}
         )
 
+    @property
+    def sorting_params(self) -> Tuple[str, SorterOps]:
+        """Converts the class variables into a list of usable Filter Models."""
+        column = self.sort_by
+        # The default sorting operand is asc
+        operator = 'asc'
```
Comment (schustmi) [inline src/zenml/models/filter_models.py:325]:
```suggestion
        operator = SorterOps.ASCENDING
```

**Review by schustmi — CHANGES_REQUESTED** [review]

File: src/zenml/models/filter_models.py (hunk 2 above)
Comment (schustmi) [reply src/zenml/models/filter_models.py:333]: But the `if` condition can be removed right? It already validates that `split_value[0]` is in `SorterOps.values()`

Comment (AlexejPenner) [reply src/zenml/models/filter_models.py]: You are of course correct, i removed it

**Review by schustmi — APPROVED** [review]
🦭

File: src/zenml/models/filter_models.py [hunk 4]
```diff
@@ -278,12 +278,27 @@ class BaseFilterModel(BaseModel):
...
+        """Validate that the sort_column is a valid column with a valid operand."""
+        column = v
+        split_value = v.split(":", 1)
+        if len(split_value) == 2:
+            column = split_value[1]
+
+            if split_value[0] not in SorterOps.values():
+                logger.warning(
+                    "Invalid operand used for column sorting. "
+                    "Only the following operands are supgitported `%s`. "
```
Comment (schustmi) [inline src/zenml/models/filter_models.py:290]: subgitported :P

**Review by schustmi — CHANGES_REQUESTED** [review]

**Review by fa9r — CHANGES_REQUESTED** [review]
Some unit tests for the new asc/desc sorting functionality would be, otherwise LGTM

File: tests/unit/models/test_filter_models.py [hunk 5]
```diff
@@ -107,6 +107,43 @@ def test_filter_model_sort_by_for_existing_field_succeeds(
...
+
+
+@pytest.mark.parametrize(
+    "correct_sortable_column",
+    [("pancakes", "created"),
+     ("", "updated"),
+     (1, "id")]
+)
+def test_filter_model_sort_by_existing_field_wrong_order_succeeds(
+    correct_sortable_column: Any,
```
Comment (schustmi) [inline tests/unit/models/test_filter_models.py:139]: Shouldn't this be `incorrect`?

**Review by schustmi — CHANGES_REQUESTED** [review]

**Review by fa9r — APPROVED** [review]
let's go to main 🚀

File: tests/unit/models/test_filter_models.py (hunk 5 above)
Comment (AlexejPenner) [reply tests/unit/models/test_filter_models.py:139]: it throws a wwarning and defaults to default order

**Pull request merged**